nova/tests/unit/compute/test_utils.py

```diff
@@ -1446,6 +1446,80 @@ def test_upsize_quota_delta_quota_separate_both_different(self):
         deltas = compute_utils.upsize_quota_delta(new_flavor, old_flavor)
         self.assertEqual(expected_deltas, deltas)
 
+    def test_upsize_quota_delta_towards_hw_version(self):
+        """Resizing from not having a hw_version towards a hw_version
+        needs all resources in addittion
+        """
+        old_flavor = objects.Flavor(vcpus=4, memory_mb=1024, extra_specs={})
+        new_flavor = objects.Flavor(
+            vcpus=5, memory_mb=2048,
+            extra_specs={utils.QUOTA_HW_VERSION_KEY: '2'})
+
+        expected_deltas = {
+            'hw_version_2_cores': new_flavor['vcpus'],
+            'hw_version_2_ram': new_flavor['memory_mb']
+        }
+
+        deltas = compute_utils.upsize_quota_delta(new_flavor, old_flavor)
+        self.assertEqual(expected_deltas, deltas)
+
+    def test_upsize_quota_delta_towards_different_hw_version(self):
+        """Resizing from one hw_version towards another hw_version
+        needs all resources in addittion
+        """
+        old_flavor = objects.Flavor(
+            vcpus=4, memory_mb=1024,
+            extra_specs={utils.QUOTA_HW_VERSION_KEY: '3'})
+        new_flavor = objects.Flavor(
+            vcpus=5, memory_mb=2048,
+            extra_specs={utils.QUOTA_HW_VERSION_KEY: '2'})
+
+        expected_deltas = {
+            'hw_version_2_cores': new_flavor['vcpus'],
+            'hw_version_2_ram': new_flavor['memory_mb']
+        }
+
+        deltas = compute_utils.upsize_quota_delta(new_flavor, old_flavor)
+        self.assertEqual(expected_deltas, deltas)
+
+    def test_upsize_quota_delta_towards_different_hw_version_less(self):
+        """Resizing from one hw_version towards another hw_version
+        needs all resources in addittion
+        """
+        old_flavor = objects.Flavor(
+            vcpus=4, memory_mb=1024,
+            extra_specs={utils.QUOTA_HW_VERSION_KEY: '244'})
```

Comment:
This VERSION_KEY distracted me from the fact that the important part of this test case vs the previous one is the reduction of resources, not the hw_version. Maybe use a less conspicuous one, or even the same as above.